### PR TITLE
add unixsocket without real inode

### PR DESCRIPTION
**Pull request by lhw2002426**
add unix socket ans AF_UNIX type for socket
but not impl SOCK_DGRAM, and use alloced index rather than inode for unixsocket because inode is not impl yet

### CODE REVIEW

File: crates/axfs_ramfs/src/file.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
- *   You can use this software according to the terms and conditions of the Mulan PSL v2.
- *   You may obtain a copy of Mulan PSL v2 at:
- *               http://license.coscl.org.cn/MulanPSL2
- *   THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
- *   See the Mulan PSL v2 for more details.
- */
+*   [Ruxos] is licensed under Mulan PSL v2.
+*   You can use this software according to the terms and conditions of the Mulan PSL v2.
+*   You may obtain a copy of Mulan PSL v2 at:
+*               http://license.coscl.org.cn/MulanPSL2
```
Comment (ken4647) [inline crates/axfs_ramfs/src/file.rs:5]: why change here?

File: api/ruxos_posix_api/src/imp/net.rs [hunk 2]
```diff
@@ -1,29 +1,53 @@
...
-use ruxfdtable::{FileLike, RuxStat};
-use ruxnet::{TcpSocket, UdpSocket};
+use ruxfdtable::{FileLike, RuxStat, RUX_FILE_LIMIT};
+use ruxnet::{SockaddrUn, TcpSocket, UdpSocket, UnixSocket, UnixSocketType};
 
 use crate::ctypes;
 use crate::utils::char_ptr_to_str;
 
+/*impl From<*const ctypes::sockaddr_un> for SockaddrUn {
+    fn from(addr: *const ctypes::sockaddr_un) -> Self {
```
Comment (ken4647) [inline api/ruxos_posix_api/src/imp/net.rs:26]: don't commit unused comments.

File: api/arceos_posix_api/src/ctypes_gen.rs [hunk 3]
```diff
@@ -0,0 +1,3458 @@
+/* automatically generated by rust-bindgen 0.66.1 */
```
Comment (ken4647) [inline api/arceos_posix_api/src/ctypes_gen.rs:1]: why in `arceos_posix_api`?

File: api/ruxos_posix_api/src/imp/net.rs [hunk 4]
```diff
@@ -1,29 +1,45 @@
...
+fn addrun_convert(addr: *const ctypes::sockaddr_un) -> SockaddrUn {
+    unsafe {
+        SockaddrUn {
+            sun_family: (*addr).sun_family,
+            sun_path: (*addr).sun_path,
+        }
+    }
+}
+
+pub enum Sockaddr {
```
[the remark this answers is not in the document]
Comment (lhw2002426) [reply api/ruxos_posix_api/src/imp/net.rs:34]: I think SockaddrUn is clear enough to represent unx socket address, and i just  change it into SockAddrUn

**Review by ken4647 — APPROVED** [review]

**Pull request merged**